src/Courant/Courant_presets.jl

```diff
@@ -0,0 +1,71 @@
+module Courant
+
+export advective_courant, diffusive_courant, nondiffusive_courant
+
+using LinearAlgebra, StaticArrays
+using ..VariableTemplates
+using ..MoistThermodynamics
+using ..PlanetParameters
+import ..MoistThermodynamics: internal_energy
+using ..SubgridScaleParameters
+using GPUifyLoops
+using CLIMA.Atmos: AtmosModel, soundspeed
+import CLIMA.DGmethods: BalanceLaw, vars_aux, vars_state,
+                        vars_diffusive
+using CLIMA.Mesh.Topologies
+using CLIMA.Mesh.Grids
+using CLIMA.Mesh.Grids: VerticalDirection, HorizontalDirection, EveryDirection
+using Logging
+using Printf
+using LinearAlgebra
+using CLIMA.Atmos
+
+
+function advective_courant(m::BalanceLaw, state::Vars, aux::Vars,
+                               diffusive::Vars, Δx, Δt, direction=VerticalDirection())
+  k̂ = aux.orientation.∇Φ / norm(aux.orientation.∇Φ)
+  if direction isa VerticalDirection
+    u = dot(state.ρu/state.ρ, k̂) * k̂
+          return Δt * (norm(u)) / Δx
+  elseif direction isa HorizontalDirection
+    u = cross(k̂, cross(state.ρu/state.ρ,k̂))
+    return Δt * (norm(u)) / Δx
+  end
+end
+
+function nondiffusive_courant(m::BalanceLaw, state::Vars, aux::Vars,
+                               diffusive::Vars, Δx, Δt, direction=VerticalDirection())
+  k̂ = aux.orientation.∇Φ / norm(aux.orientation.∇Φ)
+  if direction isa VerticalDirection
+    u = dot(state.ρu/state.ρ, k̂) * k̂
+          return Δt * (norm(u) + soundspeed(m.moisture, m.orientation, state,aux)) / Δx
+  elseif direction isa HorizontalDirection
+    u = cross(k̂, cross(state.ρu/state.ρ,k̂))      
+    return Δt * (norm(u) + soundspeed(m.moisture, m.orientation, state,
+                                            aux)) / Δx
+  end
+end
+
+function diffusive_courant(m::BalanceLaw, state::Vars, aux::Vars,
+                               diffusive::Vars, Δx, Δt, direction=VerticalDirection())
```

Comment:
@slifer50 Why can't we just define this function as `Δt * ν / Δx^2`? the `direction` is handled via the `Δx`.

Reply:
the delta x only accounts for the distance in the direction but not the orientation for the box this is unimportant but for the sphere it's important we project our viscosity in the "Horizontal" and "Vertical" directions to get the right numbers.

Reply:
ah ok, so it's to deal with cases where `ν` is not a scalar?

Reply:
Yes exactly

Reply:
Support for both isotropic and anisotropic ν added.